src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.c

```diff
@@ -393,6 +393,158 @@ ucc_status_t ucc_tl_mlx5_mcast_setup_qps(ucc_tl_mlx5_mcast_coll_context_t *ctx,
     return UCC_OK;
 }
 
+ucc_status_t ucc_tl_mlx5_mcast_create_rc_qps(ucc_tl_mlx5_mcast_coll_context_t *ctx,
+                                             ucc_tl_mlx5_mcast_coll_comm_t *comm)
+{
+    struct ibv_srq_init_attr srq_init_attr;
+    struct ibv_qp_init_attr  qp_init_attr;
+    int                      i, j;
+
+    /* create srq for this RC connection */
+    memset(&srq_init_attr, 0, sizeof(srq_init_attr));
+    srq_init_attr.attr.max_wr  = comm->params.rx_depth;
+    srq_init_attr.attr.max_sge = 2;
+
+    comm->srq = ibv_create_srq(ctx->pd, &srq_init_attr);
+    if (!comm->srq) {
+        tl_error(ctx->lib, "ibv_create_srq() failed\n");
+        return UCC_ERR_NO_RESOURCE;
+    }
+
+    comm->mcast.rc_qp = ucc_calloc(1, comm->commsize * sizeof(struct ibv_qp *), "ibv_qp* list");
+    if (!comm->mcast.rc_qp) {
+        tl_error(ctx->lib, "failed to allocate memory for ibv_qp*");
+        return UCC_ERR_NO_MEMORY;
+    }
+
+    /* create RC qp */
+    for (i = 0; i < comm->commsize; i++) {
+
+        memset(&qp_init_attr, 0, sizeof(qp_init_attr));
+
+        qp_init_attr.srq                 = comm->srq;
+        qp_init_attr.qp_type             = IBV_QPT_RC;
+        qp_init_attr.send_cq             = comm->scq;
+        qp_init_attr.recv_cq             = comm->rcq;
+        qp_init_attr.sq_sig_all          = 0;
+        qp_init_attr.cap.max_send_wr     = comm->params.sx_depth;
+        qp_init_attr.cap.max_recv_wr     = 0; // has srq
+        qp_init_attr.cap.max_inline_data = 0;
+        qp_init_attr.cap.max_send_sge    = comm->params.sx_sge;
+        qp_init_attr.cap.max_recv_sge    = comm->params.rx_sge;
+
+        comm->mcast.rc_qp[i] = ibv_create_qp(ctx->pd, &qp_init_attr);
+        if (!comm->mcast.rc_qp[i]) {
+            tl_error(ctx->lib, "Failed to create mcast RC qp index %d, errno %d", i, errno);
+            goto destroy_qp;
+        }
+    }
+
+    return UCC_OK;
+
+destroy_qp:
+    for (j=0; j<i-1; j++) {
```

Comment:
correct me if im wrong but I think it's
```suggestion
    for (j=0; j<i; j++) {
```